app/javascript/components/pame/PameModal.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+  <div 
+    :class="['modal-wrapper', { 'modal--active' : isActive }]"
+    @click.self="closeModal()"
+  >
+    <div 
+      class="modal-overlay" 
+    />
+    <div 
+      class="modal--pame"
+      id="modal"
+      :style="styleObject" 
+    >
+      <div 
+        class="modal__content"
+      >
+        <button 
+          class="modal__close" 
+          @click="closeModal()"
+        />
+
+        <h2 class="modal__title">Source details</h2>
+
+        <template v-if="hasContent(modalContent.metadata_id)">
+          <p><strong>MetadataID:</strong> {{ modalContent.metadata_id }}</p>
+        </template>
+
+        <template v-if="hasContent(modalContent.data_title)">
+          <p><strong>Data title:</strong> {{ modalContent.data_title }}</p>
+        </template>
+
+        <template v-if="hasContent(modalContent.resp_party)">
+          <p><strong>Responsible party:</strong> {{ modalContent.resp_party }}</p>
+        </template>
+
+        <template v-if="hasContent(modalContent.year)">
+          <p><strong>Year of submission:</strong> {{ modalContent.source_year }}</p>
+        </template>
+
+        <template v-if="hasContent(modalContent.language)">
+          <p><strong>Language:</strong> {{ modalContent.language }}</p>
+        </template>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+  export default {
+    name: 'modal',
+
+    data () {
+      return {
+        isActive: false,
+        modalOffset: 0,
+        styleObject: {
+          top: 0
+        },
+        modalContent: this.$store.state.pame.modalContent
```

Comment:
generally recommended to have this as computed, probably this is fine (or maybe even better) if it will never change? How come it's in the store?